src/protocol.cpp

```diff
@@ -113,31 +113,29 @@ std::string CMessageHeader::GetCommand() const
     return std::string(pchCommand, pchCommand + strnlen(pchCommand, COMMAND_SIZE));
 }
 
-bool CMessageHeader::IsValid(const MessageStartChars& pchMessageStartIn) const
+std::string CMessageHeader::GetMessageStart() const
 {
-    // Check start string
-    if (memcmp(pchMessageStart, pchMessageStartIn, MESSAGE_START_SIZE) != 0)
-        return false;
+    return std::string(pchMessageStart, pchMessageStart + strnlen(pchMessageStart, MESSAGE_START_SIZE));
```

Comment:
Nit: Keeping in the spirit of the original comment here, maybe add a comment saying returning staring string?

Reply:
The diff algorithm is confused, there wasn't ever a comment here.  This check is now in ReadHeader, where I've preserved the comment.